src/pythainer/runners/__init__.py

```diff
@@ -229,13 +229,21 @@ def get_str_command(self) -> str:
         result = " ".join(cmd)
         return result
 
-    def run(self) -> None:
+    def run(self, commands: list[str] = None) -> None:
         """
         Executes the constructed Docker command.
+
+        Parameters:
+            commands (list[str]): The commands to run before entering
+                                  into the docker in interactive mode
         """
         command = self.get_command()
         if "$@" == command[-1]:
             command = command[:-1]
+        if commands != []:
```

Comment:
```suggestion
        if commands:
```
also cover the case where it is none